pkg/operator2/starter.go

```diff
@@ -37,116 +40,171 @@ const (
+		informerToRunFn(ctx.Done())
+	}
+
+	for _, controllerRunFn := range opCtx.controllersToRunFunc {
+		go controllerRunFn(ctx, 1)
+	}
+
+	<-ctx.Done()
+	return nil
+}
+
+func prepareOauthOperator(ctx context.Context, controllerContext *controllercmd.ControllerContext, operatorCtx *operatorContext) error {
 	// protobuf can be used with non custom resources
-	oauthClient, err := oauthclient.NewForConfig(controllerContext.ProtoKubeConfig)
+	// protobuf can be used with non custom resources
+	routeClient, err := routeclient.NewForConfig(controllerContext.ProtoKubeConfig)
 	if err != nil {
 		return err
 	}
 
-	configClient, err := configclient.NewForConfig(controllerContext.KubeConfig)
+	// protobuf can be used with non custom resources
+	oauthClient, err := oauthclient.NewForConfig(controllerContext.ProtoKubeConfig)
 	if err != nil {
 		return err
 	}
 
-	kubeInformersNamespaced := informers.NewSharedInformerFactoryWithOptions(kubeClient, resync,
+	kubeInformersNamespaced := informers.NewSharedInformerFactoryWithOptions(operatorCtx.kubeClient, resync,
 		informers.WithNamespace("openshift-authentication"),
 	)
 
-	kubeSystemNamespaceInformers := informers.NewSharedInformerFactoryWithOptions(kubeClient, resync,
+	kubeSystemNamespaceInformers := informers.NewSharedInformerFactoryWithOptions(operatorCtx.kubeClient, resync,
 		informers.WithNamespace("kube-system"),
 	)
 
-	// short resync period as this drives the check frequency when checking the .well-known endpoint. 20 min is too slow for that.
-	authOperatorConfigInformers := authopinformer.NewSharedInformerFactoryWithOptions(authConfigClient, time.Second*30,
-		authopinformer.WithTweakListOptions(singleNameListOptions("cluster")),
-	)
-
 	routeInformersNamespaced := routeinformer.NewSharedInformerFactoryWithOptions(routeClient, resync,
 		routeinformer.WithNamespace("openshift-authentication"),
 		routeinformer.WithTweakListOptions(singleNameListOptions("oauth-openshift")),
 	)
 
 	// do not use WithTweakListOptions here as top level configs are all called "cluster"
 	// whereas our cluster operator instance is called "authentication" (there is no OR support)
-	configInformers := configinformer.NewSharedInformerFactoryWithOptions(configClient, resync)
-
-	resourceSyncerInformers := v1helpers.NewKubeInformersForNamespaces(
-		kubeClient,
-		"openshift-authentication",
-		"openshift-config",
-		"openshift-config-managed",
-	)
-
-	operatorClient := &OperatorClient{
-		authOperatorConfigInformers,
-		authConfigClient.OperatorV1(),
-	}
-
-	resourceSyncer := resourcesynccontroller.NewResourceSyncController(
-		operatorClient,
-		resourceSyncerInformers,
-		v1helpers.CachedSecretGetter(kubeClient.CoreV1(), resourceSyncerInformers),
-		v1helpers.CachedConfigMapGetter(kubeClient.CoreV1(), resourceSyncerInformers),
-		controllerContext.EventRecorder,
-	)
+	configInformers := configinformer.NewSharedInformerFactoryWithOptions(operatorCtx.configClient, resync)
```

Comment:
these factories should be shared. Above is another instance.

Reply:
config informers are only used and created here.